ansible_base/rbac/models/role.py

```
@@ -263,6 +263,195 @@ def give_permission(self, actor, content_object):
+                validate_assignment(rd, actor, obj)
+                has_team_perm = rd.permissions.filter(codename=permission_registry.team_permission).exists()
+                has_org_member = rd.permissions.filter(codename='member_organization').exists()
+                validate_team_assignment_enabled(obj_ct, has_team_perm=has_team_perm, has_org_member=has_org_member)
+                validated_pairs.add(key)
+            all_triples.append((rd, actor, obj, obj_ct))
+
+        # -- bulk ObjectRole creation: group by (rd_id, ct_id) --
+        from collections import defaultdict
+
+        or_groups = defaultdict(set)  # (rd_id, ct_id) -> set of object_id strings
+        for rd, actor, obj, obj_ct in all_triples:
```

Ruff (0.16.0)

[warning] Loop control variable `actor` not used within loop body

(B007)